library/proc_macro/src/bridge/client.rs

```diff
@@ -45,20 +47,25 @@ macro_rules! define_handles {
 
         $(
             #[repr(C)]
-            pub(crate) struct $oty(handle::Handle);
-            impl !Send for $oty {}
-            impl !Sync for $oty {}
```

Comment:
The rust-analyzer copies just drop these negative impls instead of using PhantomData, making the interface unsound.

Reply:
It seems like we probably actually want this to be even more general -- e.g., probably !Unpin for $oty also makes sense. I'm not sure there's a type that's truly "I implement no auto traits" currently...

Reply:
Raw pointers are neither send nor sync. They are unpin.